File: Java/Method/Odd-Even.java
import java.util.Scanner;

public class Method {

    // Method to calculate the sum of odd numbers up to n
    public static int sumOdd(int n) {
        if (n <= 0) return 0;
        if (n % 2 != 0) {
            return n + sumOdd(n - 2);
        } else {
            return sumOdd(n - 1);
        }
    }

    // Method to calculate the sum of even numbers up to n
    public static int sumEven(int n) {
        if (n <= 0) return 0;
        if (n % 2 == 0) {
            return n + sumEven(n - 2);
        } else {
            return sumEven(n - 1);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Prompting user for input
        System.out.println("Enter an integer:");

        int n = sc.nextInt();

        // Calculate sum of odd and even numbers
        int sumOdd = sumOdd(n);
        int sumEven = sumEven(n);

        // Displaying results
        System.out.println("Sum of odd numbers up to " + n + ": " + sumOdd);
        System.out.println("Sum of even numbers up to " + n + ": " + sumEven);

        sc.close();  // Close the scanner to avoid resource leaks
    }
}

File: C/Array/Odd-Even.c
#include<stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int a[1000];
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    printf("Even numbers in array: ");
    for (int i = 0; i < n; i++) {
        if (a[i] % 2 == 0) {
            printf("%d ", a[i]);
        }
    }
    printf("\nOdd numbers in array: ");
    for (int i = 0; i < n; i++) {
        if (a[i] % 2 != 0) {
            printf("%d ", a[i]);
        }
    }
    return 0;
}
